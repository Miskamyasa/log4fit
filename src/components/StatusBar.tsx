import {ReactElement, memo} from "react";
import {StyleSheet, Platform, StatusBar as StatusBarNative, View, ViewStyle} from "react-native";
import Constants from "expo-constants";


const styles: {root: ViewStyle} = StyleSheet.create({
  root: {
    height: Constants.statusBarHeight,
    backgroundColor: "#000",
  },
});

function StatusBar(): ReactElement {
  if (Platform.OS === "ios") {
    return (
      <View style={styles.root}>
        <StatusBarNative barStyle="light-content" />
      </View>
    );
  }
  return (
    <StatusBarNative barStyle="light-content" />
  );
}

export default memo(StatusBar, () => true);
