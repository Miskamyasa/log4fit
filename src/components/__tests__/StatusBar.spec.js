import renderer from "react-test-renderer";

import StatusBar from "../StatusBar";


describe("components: StatusBar", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<StatusBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
