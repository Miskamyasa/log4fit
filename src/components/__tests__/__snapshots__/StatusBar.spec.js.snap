// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: StatusBar renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#000",
      "height": 20,
    }
  }
/>
`;
